crates/client/db/src/messaging_db.rs

```diff
@@ -0,0 +1,81 @@
+use rocksdb::WriteOptions;
+use serde::{Deserialize, Serialize};
+use starknet_api::core::Nonce;
+use std::collections::BTreeSet;
+
+use crate::error::DbError;
+use crate::{Column, DatabaseExt, DeoxysBackend, DeoxysStorageError};
+
+type Result<T, E = DeoxysStorageError> = std::result::Result<T, E>;
+
+pub const LAST_SYNCED_L1_EVENT_BLOCK: &[u8] = b"LAST_SYNCED_L1_EVENT_BLOCK";
+pub const NONCES_PROCESSED: &[u8] = b"NONCES_PROCESSED";
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct LastSyncedEventBlock {
+    pub block_number: u64,
+    pub event_index: u64,
+}
+
+impl LastSyncedEventBlock {
+    pub fn new(block_number: u64, event_index: u64) -> Self {
+        LastSyncedEventBlock { block_number, event_index }
+    }
+}
+
+impl DeoxysBackend {
+    pub fn messaging_last_synced_l1_block_with_event(&self) -> Result<Option<LastSyncedEventBlock>> {
```

Comment:
Could you add some rust doc on top of your functions/structure. We aim to have a complete rust doc in the future

Reply:
overall seems correct to me thanks! @azurwastaken @EvolveArt

Reply:
Added some, tell me if its ok @antiyro !